Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -388,18 +388,36 @@ private Task<SFSession> NewSessionAsync(String connectionString, SecureString pa
                 }, TaskContinuationOptions.NotOnCanceled); 
         }
 
+        internal void ReleaseBusySession(SFSession session)
+        {
+            s_logger.Debug("SessionPool::ReleaseBusySession");
+            int currentPoolSize;
+            lock (_sessionPoolLock)
+            {
+                _busySessionsCounter.Decrease();
+                currentPoolSize = GetCurrentPoolSize();
+            }
+            var currentSizeMessageOldPool = $"After releasing a busy session from the pool, the pool size is: {currentPoolSize}";
+            var poolSizeMessage = IsMultiplePoolsVersion()
```

Comment:
WDYT about simple "Session released" and adding reusable (in multiple places) method to log pool status (counters, timers, min/max pool sizes)?

Reply:
not in this PR